frontend/src/app/page.tsx

```diff
@@ -1,12 +1,13 @@
 "use client";
 
-import { getVideos, Video } from "@/shared/api";
+import { getVideos, Progress, Video } from "@/shared/api";
 import VideoCard from "@/shared/video-card";
 import Link from "next/link";
 import { useState, useEffect } from "react";
 
 export default function Home() {
   const [videos, setVideos] = useState<Video[]>([]);
+  const [videoProgresses, setVideoProgresses] = useState<Record<string, number>>({})
```

Comment:
The `Record<string, number>` should probably be a type, since it's coming from the API